src/3d/shaders/mesh/mesh.frag

```diff
@@ -0,0 +1,275 @@
+#version 330 core
+
+in treatedVertex {
+    vec3 wordPosition;
+    vec3 wordNormal;
+    float magnitude;
+    noperspective vec4 edgeA;
+    noperspective vec4 edgeB;
+    flat int configuration;
+} fs_in;
+
+out vec4 fragColor;
+
+uniform bool wireframeEnabled;
+uniform float lineWidth;
+uniform vec4 lineColor;
+
+uniform int textureType;
+uniform int colorRampType;
+uniform vec4 meshColor;
+uniform float verticaleScale;
+
+uniform sampler1D colorRampTexture;
+uniform int colorRampCount;
+
+const int MAX_LIGHTS = 8;
+const int TYPE_POINT = 0;
+const int TYPE_DIRECTIONAL = 1;
+const int TYPE_SPOT = 2;
+struct Light {
+    int type;
+    vec3 position;
+    vec3 color;
+    float intensity;
+    vec3 direction;
+    float constantAttenuation;
+    float linearAttenuation;
+    float quadraticAttenuation;
+    float cutOffAngle;
+};
+uniform Light lights[MAX_LIGHTS];
+uniform int lightCount;
+
+
+vec4 wireframeShadeLine( const in vec4 color )
```

Comment:
it would be good to point to the original code for reference

Reply:
sure!